tools/testbench/topology.c

```diff
@@ -680,7 +607,7 @@ static int load_fileread(struct tplg_context *ctx, int dir)
 	struct sof *sof = ctx->sof;
 	struct testbench_prm *tp = ctx->tp;
 	FILE *file = ctx->file;
-	struct sof_ipc_comp_file fileread = {0};
+	struct sof_ipc_comp_file fileread = {{{0}}};
```

Comment:
I think just `struct xxx x = {};` should work - there are several locations in this PR

Reply:
I've read on stackoverflow[*] that `{}` is C++ and C23. And a gcc extension. 

I would simply not bother trying to make sense of the maze standards and stick to using one `0` element.

[*] so it must be true!

Reply:
With this change

```
--- a/tools/testbench/topology.c
+++ b/tools/testbench/topology.c
@@ -607,7 +607,7 @@ static int load_fileread(struct tplg_context *ctx, int dir)
        struct sof *sof = ctx->sof;
        struct testbench_prm *tp = ctx->tp;
        FILE *file = ctx->file;
-       struct sof_ipc_comp_file fileread = {{{0}}};
+       struct sof_ipc_comp_file fileread = {0};
        int ret;
 
        fileread.config.frame_fmt = find_format(tp->bits_in);

```

I get compile error

```
/home/singalsu/xtensa/XtDevTools/install/tools/RG-2017.8-linux/XtensaTools/bin/xt-xcc  -DRELATIVE_FILE=\"topology.c\" -I/home/singalsu/work/current/sof/sof/tools/testbench/include -I/home/singalsu/work/current/sof/sof/tools/testbench/xtos/include -I/usr/include/alsa/sound/uapi -I/home/singalsu/work/current/sof/sof/tools/testbench/build_xt_testbench/sof_ep/install/include -I/home/singalsu/work/current/sof/sof/tools/testbench/build_xt_testbench/sof_parser/install/include -g -O3 -Wall -Werror -Wl,-EL -Wmissing-prototypes -DCONFIG_LIBRARY -DCONFIG_LIBRARY_STATIC -imacros/home/singalsu/work/current/sof/sof/tools/testbench/build_xt_testbench/sof_ep/build/library_autoconfig.h -MD -MT CMakeFiles/testbench.dir/topology.c.o -MF CMakeFiles/testbench.dir/topology.c.o.d -o CMakeFiles/testbench.dir/topology.c.o -c /home/singalsu/work/current/sof/sof/tools/testbench/topology.c
cc1: warnings being treated as errors
/home/singalsu/work/current/sof/sof/tools/testbench/topology.c: In function ‘load_fileread’:
/home/singalsu/work/current/sof/sof/tools/testbench/topology.c:610: warning: missing braces around initializer
/home/singalsu/work/current/sof/sof/tools/testbench/topology.c:610: warning: (near initialization for ‘fileread.comp’)

```

If I do just `struct sof_ipc_comp_file fileread = {};` it compiles. Hope it zero initializes the struct too. Is C23 bad? Need your guidance what to do.

Reply:
Just keep `{{{0}}};`, it's great.

Since the dawn of time the C standard(s) have guaranteed that partial initialization is automatically extended and automatically behaves like static initialization. I guess the idea behind the more recent `{}` improvement was to save you from counting the number of brackets? Developer productivity breakthrough of the decade! /s

More likely: C23 just re-aligning with C++ as it usually does.